app/controllers/books_controller.rb

```diff
@@ -40,18 +37,11 @@ def create
   end
 
   def update
-    @book.authors = []
-    
-    author_params[:author_ids].each do |k, v|
-      @book.authors << Author.find(k) unless k.empty?
-    end
-    
-    @book.image.purge if delete_params[:delete_image].to_i == 1
-    
     respond_to do |format|
       if @book.update(book_params)
-        @book
-        format.html { redirect_to books_path, notice: "Book was successfully updated." }
+        @book.image.purge if delete_params[:delete_image].to_i == 1
```

Comment:
[Suggestion]
We should do the delete image after we save the book successful. If we put this code before we save the book, in the case that the book can't be saved, the image still be deleted.